components/tidb_query_vec_executors/src/table_scan_executor.rs

```diff
@@ -75,6 +75,7 @@ impl<S: Storage> BatchTableScanExecutor<S> {
             columns_default_value,
             column_id_index,
             handle_indices,
+            primary_column_ids,
```

Comment:
Better to add assertion (or return error) that when `primary_column_ids` is given, `handle_indices` is empty (and vise versa)

Reply:
Maybe we should also verify that `pk_handle` is an int column in L61 (`if ci.get_pk_handle() { ... }`)

Reply:
For `is_key_only = false` in L64, it need to be changed when corresponding pk exists in `primary_column_ids`.